pmlc/conversion/gpu/gpu_to_vulkan.cc

```diff
@@ -158,19 +175,18 @@ class ConvertGpuLaunchFuncToVulkanCalls
   LLVM::LLVMType getLLVMInt64Type() { return llvmInt64Type; }
 
   mlir::Type getMLIRFloat32Type() { return mlirFloat32Type; }
-  mlir::Type getMLIRIndexType() { return mlirIndexType; }
 
   LLVM::LLVMType llvmVoidType;
   LLVM::LLVMType llvmPointerType;
   LLVM::LLVMType llvmInt32Type;
   LLVM::LLVMType llvmInt64Type;
 
   mlir::Type mlirFloat32Type;
-  mlir::Type mlirIndexType;
 
   uint64_t numKernel = 0;
   uint64_t lauchFuncIndex = 0;
   mlir::Value vulkanRuntime;
+  LLVM::CallOp deinitVulkan;
```

Comment:
It seems like there may be a design issue here if we need an (uninitialized) member variable to communicate between two member methods --- `convertGpuLaunchFunc` which sets `deinitVulkan` and `getBufferCopyMode` which consumes it.  I see that both methods are private and that ordering of calls to these methods is preserved by `runOnOperation` so it's not a bug, technically.  Again, just wondering if the need for this sort of thing indicates some design issue.  Could we reorganize code, somehow, to avoid this problem?

Reply:
Fixed and replaced "deinitVulkan" by "run", which is clearer.